src/transformStateWithClones.js

```diff
@@ -6,8 +6,30 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stansAfterAction = [];
+  let copyState = { ...state};
+
+actions.forEach((action) => {
+if (action.type === 'addProperties') {
+Object.assign(copyState, action.extraData);
+let doubleCopyState = { ...copyState};
+stansAfterAction.push(doubleCopyState);
+}
+if (action.type === 'removeProperties') {
+  action.keysToRemove.forEach(key => delete copyState[key]);
+  let doubleCopyState = { ...copyState};
+  stansAfterAction.push(doubleCopyState);
 }
+if (action.type === 'clear') {
+  Object.keys(copyState).forEach(key => delete copyState[key]);
```

Comment:
you check in each if `action.type` values, better to use switch case for such cases

Reply:
and looks like you forgot to add spaces and tabulation

Reply:
ok, thx